### PR TITLE
Migration tpf

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
PR for migration of TPF version from 0.10 to 1.0.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Comment by rounak-adhikary** [comment]
We tested for only datasources yet
```bash
bash testdata.sh && go test -coverprofile=coverage.out terraform-provider-ome/ome -v -timeout 5h -run TestDataSource_Read
=== RUN   TestDataSource_ReadConfigurationReport
--- PASS: TestDataSource_ReadConfigurationReport (79.24s)
=== RUN   TestDataSource_ReadGroupDevices
--- PASS: TestDataSource_ReadGroupDevices (9.61s)
=== RUN   TestDataSource_ReadTemplate
--- PASS: TestDataSource_ReadTemplate (53.32s)
=== RUN   TestDataSource_ReadVlanNetworks
--- PASS: TestDataSource_ReadVlanNetworks (2.80s)
PASS
coverage: 28.1% of statements
ok      terraform-provider-ome/ome      144.998s        coverage: 28.1% of statements
go tool cover -html coverage.out -o coverage.html
```



image

**Comment by Krishnan-Priyanshu** [comment]
we've generated all the coverage report except the configuration compliance resource
image

**Comment by Krishnan-Priyanshu** [comment]
```
root@lglap049:~/terraform-provider-ome# bash testdata.sh && /usr/local/go/bin/go test terraform-provider-ome/ome -v -timeout 5h -run "TestDataSource_Read|TestTemplate|TestCreateBaseline" -coverprofile=coverage.out
=== RUN   TestDataSource_ReadConfigurationReport
--- PASS: TestDataSource_ReadConfigurationReport (79.10s)
=== RUN   TestDataSource_ReadGroupDevices
--- PASS: TestDataSource_ReadGroupDevices (8.94s)
=== RUN   TestDataSource_ReadTemplate
--- PASS: TestDataSource_ReadTemplate (53.11s)
=== RUN   TestDataSource_ReadVlanNetworks
--- PASS: TestDataSource_ReadVlanNetworks (3.06s)
=== RUN   TestCreateBaseline_TestValidations
--- PASS: TestCreateBaseline_TestValidations (10.32s)
=== RUN   TestCreateBaseline_BaselineWithDeviceID
--- PASS: TestCreateBaseline_BaselineWithDeviceID (154.00s)
=== RUN   TestCreateBaseline_CreateBaselineWithSchedule
--- PASS: TestCreateBaseline_CreateBaselineWithSchedule (113.19s)
=== RUN   TestCreateBaseline_CreateBaselineWithScheduleNonCompliant
--- PASS: TestCreateBaseline_CreateBaselineWithScheduleNonCompliant (110.28s)
=== RUN   TestTemplateDeploy_InvalidTemplate
--- PASS: TestTemplateDeploy_InvalidTemplate (2.69s)
=== RUN   TestTemplateDeploy_CreateAndUpdateDeploySuccess
--- PASS: TestTemplateDeploy_CreateAndUpdateDeploySuccess (193.26s)
=== RUN   TestTemplateDeploy_CreateUpdateDeployWithScheduleSuccess
--- PASS: TestTemplateDeploy_CreateUpdateDeployWithScheduleSuccess (49.77s)
=== RUN   TestTemplateDeploy_ImportDeploymentError
--- PASS: TestTemplateDeploy_ImportDeploymentError (116.88s)
=== RUN   TestTemplateDeploy_CreateDeployBootNetworkISOSuccess
--- PASS: TestTemplateDeploy_CreateDeployBootNetworkISOSuccess (251.58s)
=== RUN   TestTemplateCreation_CreateAndUpdateTemplateSuccess
--- PASS: TestTemplateCreation_CreateAndUpdateTemplateSuccess (192.14s)
=== RUN   TestTemplateCreation_CreateTemplateByCloningSuccess
--- PASS: TestTemplateCreation_CreateTemplateByCloningSuccess (178.63s)
=== RUN   TestTemplateCreation_CreateTemplatesInvalidScenarios
--- PASS: TestTemplateCreation_CreateTemplatesInvalidScenarios (6.60s)
=== RUN   TestTemplateImport_ImportTemplates
--- PASS: TestTemplateImport_ImportTemplates (88.99s)
=== RUN   TestTemplateCreation_CreateImportTemplate
--- PASS: TestTemplateCreation_CreateImportTemplate (11.55s)
PASS
coverage: 67.6% of statements
ok      terraform-provider-ome/ome      1624.152s       coverage: 67.6% of statements
root@lglap049:~/terraform-provider-ome#```

**Comment by Krishnan-Priyanshu** [comment]
```
root@lglou045:~/priyanshu/terraform-provider-ome# go test -timeout 90m -coverprofile=coverage_all.out terraform-provider-ome/ome -v
=== RUN   TestDataSource_ReadConfigurationReport
--- PASS: TestDataSource_ReadConfigurationReport (78.72s)
=== RUN   TestDataSource_ReadGroupDevices
--- PASS: TestDataSource_ReadGroupDevices (8.99s)
=== RUN   TestDataSource_ReadTemplate
--- PASS: TestDataSource_ReadTemplate (52.33s)
=== RUN   TestDataSource_ReadVlanNetworks
--- PASS: TestDataSource_ReadVlanNetworks (2.60s)
=== RUN   TestCreateBaseline_TestValidations
--- PASS: TestCreateBaseline_TestValidations (6.74s)
=== RUN   TestCreateBaseline_BaselineWithDeviceIDAndTags
--- PASS: TestCreateBaseline_BaselineWithDeviceIDAndTags (271.91s)
=== RUN   TestCreateBaseline_CreateBaselineWithSchedule
--- PASS: TestCreateBaseline_CreateBaselineWithSchedule (111.93s)
=== RUN   TestCreateBaseline_CreateBaselineWithScheduleNonCompliant
--- PASS: TestCreateBaseline_CreateBaselineWithScheduleNonCompliant (110.62s)
=== RUN   TestConfigurationRemediationErrors
--- PASS: TestConfigurationRemediationErrors (84.65s)
=== RUN   TestConfigurationRemediation
--- PASS: TestConfigurationRemediation (288.71s)
=== RUN   TestTemplateDeploy_InvalidTemplate
--- PASS: TestTemplateDeploy_InvalidTemplate (2.40s)
=== RUN   TestTemplateDeploy_CreateAndUpdateDeploySuccess
--- PASS: TestTemplateDeploy_CreateAndUpdateDeploySuccess (193.84s)
=== RUN   TestTemplateDeploy_CreateUpdateDeployWithScheduleSuccess
--- PASS: TestTemplateDeploy_CreateUpdateDeployWithScheduleSuccess (49.30s)
=== RUN   TestTemplateDeploy_ImportDeploymentError
--- PASS: TestTemplateDeploy_ImportDeploymentError (116.50s)
=== RUN   TestTemplateDeploy_CreateDeployBootNetworkISOSuccess
--- PASS: TestTemplateDeploy_CreateDeployBootNetworkISOSuccess (252.20s)
=== RUN   TestTemplateCreation_CreateAndUpdateTemplateSuccess
--- PASS: TestTemplateCreation_CreateAndUpdateTemplateSuccess (233.90s)
=== RUN   TestTemplateCreation_CreateTemplateByCloningSuccess
--- PASS: TestTemplateCreation_CreateTemplateByCloningSuccess (182.27s)
=== RUN   TestTemplateCreation_CreateTemplatesInvalidScenarios
--- PASS: TestTemplateCreation_CreateTemplatesInvalidScenarios (5.42s)
=== RUN   TestTemplateImport_ImportTemplates
--- PASS: TestTemplateImport_ImportTemplates (88.60s)
=== RUN   TestTemplateCreation_CreateImportTemplate
--- PASS: TestTemplateCreation_CreateImportTemplate (10.49s)
PASS
coverage: 77.8% of statements
ok      terraform-provider-ome/ome      2152.140s       coverage: 77.8% of statements
```

**Comment by Krishnan-Priyanshu** [comment]
image

**Review by rounak-adhikary — DISMISSED** [review]

File: coverage.html [hunk 1]
```diff
@@ -0,0 +1,5561 @@
+
+<!DOCTYPE html>
```
Comment (AnikaAgiwal2711) [inline coverage.html:2]: we don't need this file. you can add this to gitignore

Comment (Krishnan-Priyanshu) [reply coverage.html]: @AnikaAgiwal2711 we have removed the file and added the coverage out for adding up more acceptance test

**Review by shenda1 — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
LGTM

**Pull request merged**